gateleen-security/src/main/java/org/swisspush/gateleen/security/authorization/Authorizer.java

```diff
@@ -67,47 +63,42 @@ public Authorizer(Vertx vertx, final ResourceStorage storage, String securityRoo
         this.aclRoot = securityRoot + aclKey + "/";
         this.aclUriPattern = new PatternHolder(Pattern.compile("^" + aclRoot + "(?<role>.+)$"));
         this.userUriPattern = Pattern.compile(securityRoot + "user(\\?.*)?");
+        this.roleMapperUriPattern = new PatternHolder(Pattern.compile("^" + securityRoot + "/rolemapper"));
         this.roleExtractor = new RoleExtractor(rolePattern);
-
-        this.aclSchema = ResourcesUtils.loadResource("gateleen_security_schema_acl", true);
-        this.aclFactory = new AclFactory(aclSchema);
+        this.roleMapper = new RoleMapper(storage,securityRoot);
+        this.roleAuthorizer = new RoleAuthorizer(storage,securityRoot,rolePattern,roleMapper);
+        this.aclFactory = new AclFactory();
 
         eb = vertx.eventBus();
 
-        initialGrantedRoles = new HashMap<>();
-        initialGrantedRoles.put(aclUriPattern, new HashMap<>());
-        initialGrantedRoles.get(aclUriPattern).put("PUT", new HashSet<>());
-        initialGrantedRoles.get(aclUriPattern).put("GET", new HashSet<>());
-        initialGrantedRoles.get(aclUriPattern).put("DELETE", new HashSet<>());
-        initialGrantedRoles.get(aclUriPattern).get("PUT").add(adminRole);
-        initialGrantedRoles.get(aclUriPattern).get("GET").add(adminRole);
-        initialGrantedRoles.get(aclUriPattern).get("DELETE").add(adminRole);
-
-        updateAllAcls();
-
         // Receive update notifications
-        eb.consumer(UPDATE_ADDRESS, (Handler<Message<String>>) role -> updateAllAcls());
+        eb.consumer(UPDATE_ADDRESS, (Handler<Message<String>>) role -> updateAllConfigs());
     }
 
     @Override
     public void enableResourceLogging(boolean resourceLoggingEnabled) {
         this.logACLChanges = resourceLoggingEnabled;
     }
 
-    public Future<Boolean> authorize(final HttpServerRequest request){
+
+    public Future<Boolean> authorize(final HttpServerRequest request) {
         Future<Boolean> future = Future.future();
 
         handleUserUriRequest(request, future);
 
-        if(!future.isComplete()){
-            handleIsAuthorized(request, future);
+        if (!future.isComplete()) {
+            roleAuthorizer.handleIsAuthorized(request, future);
         }
 
-        if(!future.isComplete()){
-            handleAclUriRequest(request, future);
+        if (!future.isComplete()) {
+            handleConfigurationUriRequest(request, future, aclUriPattern, roleAuthorizer);
         }
 
-        if(!future.isComplete()) {
+        if (!future.isComplete()) {
+            handleConfigurationUriRequest(request, future, roleMapperUriPattern, roleMapper);
```

Comment:
Hmmm.. The `future` will be resolved async (if at all) - but you check 'now' if the `future.isComplete()`.
The whole pattern to work with a future here seems a little bit strange... Why not simply return a boolean which say 'true' or 'false'?

Reply:
Huuuu, this was already like this. Didn't touch it really.
Probably done like this because the caller of the method uses the future for further treatment ..

Reply:
I propose to leave it as is. Would cause again major changes out of scope of the initial task.